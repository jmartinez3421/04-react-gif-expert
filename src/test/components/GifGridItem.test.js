import React from 'react'
import "@testing-library/jest-dom"
import { shallow } from 'enzyme'
import GifGridItem from '../../components/GifGridItem'

describe('Pruebas sobre el componente GifGridItem', () => {
    
    const url = 'https://media3.giphy.com/media/xT9DPpf0zTqRASyzTi/giphy.gif';
    const title = 'darth vader father GIF by Star Wars';
    const wrapper = shallow(<GifGridItem url={url} title={title} />);

    test('El componente debe verse correctamente', () => {      

        expect(wrapper).toMatchSnapshot();
    });

    test('debe haber un parrafo con el titulo', () => {
        const p = wrapper.find('p');

        expect(p.text().trim()).toBe(title);
    });

    test('debe tener la imagen con la url y el title igual que los props', () => {
        const img = wrapper.find('img');

        const {src, alt} = img.props();

        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    test('el div debe tener la clase animate__fadeIn', () => {
        const div = wrapper.find('div');

        const {className} = div.props();

        expect(className).toContain('animate__fadeIn');
        expect(className.includes('animate__fadeIn')).toBe(true);
    })
})