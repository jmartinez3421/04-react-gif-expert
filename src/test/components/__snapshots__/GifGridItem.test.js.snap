// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el componente GifGridItem El componente debe verse correctamente 1`] = `
<div
  className="card animate__animated animate__fadeInDown"
>
  <img
    alt="darth vader father GIF by Star Wars"
    src="https://media3.giphy.com/media/xT9DPpf0zTqRASyzTi/giphy.gif"
  />
  <p>
    darth vader father GIF by Star Wars
  </p>
</div>
`;
